### ISSUE TITLE
[GlobalStep] "Missing Dependencies" error triggered after clicking "Activate" for the recent "Release Candidate" for WooCommerce on "Jetpack > Beta Tester" page.

### ISSUE THREAD

**Issue by gglobalstep** [woocommerce/woocommerce#53891]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

"Missing Dependencies" error triggered after clicking "Activate" for the recent "Release Candidate" for WooCommerce on "Jetpack > Beta Tester" page.

### Expected behavior

Upon clicking the "Activate" button for the recent "Release Candidate" for WooCommerce the user should be able to successfully activate the plugin without encountering any errors on "Jetpack > Beta Tester" page.

### Actual behavior

"Missing Dependencies" error triggered after clicking "Activate" for the recent "Release Candidate" for WooCommerce on "Jetpack > Beta Tester" page.

### Steps to reproduce

 1. Create a test site.
 2. Install and activate all the required plugins.
 3. Install and activate the Woocommerce version 9.6.0-beta.1 plugins.
 4. Access the WP admin of the newly created site, click on the WooCommerce menu item, and you should be taken to the WooCommerce profiler. Click "Skip guided setup" and choose US as the store's base location country.
 5. Go to Tools > WCA Test Helper > Features and enable the reactify-classic-payments-settings feature.
 6. Click on the Payments main menu item and you should see the Payments settings page with WooPayments and PayPal as payment extensions suggestions (PESs) in the main provider list and some others in the Others section
 7. Click on "Take offline payments" in the providers list and you should see a page with the offline payment methods options, with their icons properly showing.
 8. Go to Jetpack > Beta Tester and click on the Manage button for "WooCommerce Payments".
 9. On the next page, "Activate" the most recent "Release Candidate" for WooPayments.
10. Go to Plugins, and when you hover of the "Deactivate" link for WooPayments, you will see that the URL contains the plugin param with the value "woocommerce-payments-dev%2Fwoocommerce-payments.php".
11. Go to Jetpack > Beta Tester and click on the Manage button for "WooCommerce".
12. On the next page, "Activate" the most recent "Release Candidate" for "WooCommerce".
13. Observe that, "Missing Dependencies" error triggered after clicking "Activate" for the recent "Release Candidate" for WooCommerce on "Jetpack > Beta Tester" page.

### Instruction Link:

[https://github.com/woocommerce/woocommerce/pull/53577](<https://github.com/woocommerce/woocommerce/pull/53577>)

### Video:

https://github.com/user-attachments/assets/530319e3-b542-444e-b3bc-d282909ec96d

![Image](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/078b38f9-bf6f-45a4-82a8-77c80a764487/101d33ea-2bdf-4339-bba8-a97aa9c3a0cc?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi8wNzhiMzhmOS1iZjZmLTQ1YTQtODJhOC03N2M4MGE3NjQ0ODcvMTAxZDMzZWEtMmJkZi00MzM5LWJiYTgtYTk3YWE5YzNhMGNjIiwiaWF0IjoxNzg1NTEwNzc0LCJleHAiOjE4MTcwODEzMzR9.qpsLJwSwPkcq4TNclNArqEG-OkixpYG-mn0EnmWqr7E)

### WordPress Environment

Woocommerce Version: 9.6.0-beta.1<br>WordPress version: 6.7.2<br>PC:<br>Windows 10<br>Chrome (Version 131.0.6778.205)<br>Firefox (Version 133.0.3)

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](<https://woocommerce.com/products/storefront/>).
- [X] I can reproduce this bug consistently using the steps above.

System Status Report

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.6.0-beta.1
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.9.0
Log Directory Writable: ✔
WP Version: 6.7.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.2.26
PHP Post Max Size: 2 GB
PHP Time Limit: 850
PHP Max Input Vars: 6144
cURL Version: 8.10.1
OpenSSL/1.1.1w

SUHOSIN Installed: –
MySQL Version: 10.6.19-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 1
page: 7
post: 2
revision: 1
wp_font_face: 36
wp_font_family: 12
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (5) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.5
Jetpack Beta Tester: by Automattic – 4.1.1
Query Monitor: by John Blackbourn – 3.17.0
WooCommerce Beta Tester: by WooCommerce – 2.5.1
WooCommerce: by Automattic – 9.6.0-beta.1

### Inactive Plugins (2) ###

WooCommerce: by Automattic – 8.9.0-dev-8482717630-g9a50fca
WooPayments: by WooCommerce – 8.7.0-dev_PR-9977_commit-96cec56

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
db.php: Query Monitor Database Class (Drop-in)
object-cache.php: Memcached

### Must Use Plugins (1) ###

load.php: by  –

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 10 KB

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/ -  Contains the woocommerce/cart block
Checkout: #7 - /checkout/ -  Contains the woocommerce/checkout block
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.3
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
Theme type: Block theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/error.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store
reactify-classic-payments-settings

Disabled Features: product-data-views
experimental-blocks
coming-soon-newsletter-template
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
use-wp-horizon
add-to-cart-with-options-stepper-layout
blockified-add-to-cart
beta-tester-slotfill-examples

Daily Cron: ✔ Next scheduled: 2024-12-20 12:51:21 +00:00
Options: ✔
Notes: 57
Onboarding: skipped

### Action Scheduler ###

Complete: 8
Oldest: 2024-12-19 12:51:26 +0000
Newest: 2024-12-19 13:17:22 +0000

Pending: 2
Oldest: 2024-12-20 00:57:21 +0000
Newest: 2024-12-20 12:51:26 +0000


### Status report information ###

Generated at: 2024-12-19 13:51:27 +00:00
```

**Comment by rodelgc** [comment]
I'm not sure whether the root cause of the issue is in the WooPayments or WC Beta Tester, but since the issue was discovered from a payments-related PR, I'll let the payment teams take a look first.

**Comment by vbelolapotkov** [comment]
Thanks for reporting! Forwarding to @anu-rock and his team.

**Comment by elazzabi** [comment]
I can't reproduce this with a fresh site. @gglobalstep can you please double-check if this is still happening with a new site?

**Comment by anu-rock** [comment]
Hi @gglobalstep, I have the following observations. Could you help me with the answers?

1. The steps to reproduce do not match the video walkthrough. What is the source of truth here?
2. In the video, Jetpack Beta Tester was used to install **WooCommerce** RC rather than **WooPayments** RC as instructed in the linked PR. Was this intentional or an oversight?

P.S. I'm assuming `gglobalstep` is the right GH handle to ping. If not, kindly let us know the correct one.

**Comment by gglobalstep** [comment]
Hi @rodelgc @anu-rock @elazzabi, Apologies for the delay in responding,

1. Here is the video that provides step-by-step instructions for reproducing the issue:

https://github.com/user-attachments/assets/f6d66b57-7719-456f-b307-dc80174cfc2b

2. As per the PR [#53577](https://github.com/woocommerce/woocommerce/pull/53577) , we found this error on step 9 where it is mention to do the same steps for WooCommerce which were done for WooPayments in previous steps (6 - 7)

3. "Missing Dependencies" error triggered only after clicking "Activate" button for the recent "Release Candidate" for WooCommerce on "Jetpack > Beta Tester" page.

4. After clicking "Activate" button for the recent "Release Candidate" for WooCommerce 9.6.0-beta.1, the version is changed to WooCommerce 8.9.0-dev.

![Image](https://github.com/user-attachments/assets/3d5009d4-f4d5-40b2-ba69-0b4633b3c028)

Here are the logs for new site:

<details>

`
### WordPress Environment ###

WC Version: 8.9.0
REST API Version: ✔ 8.9.0
Action Scheduler Version: ✔ 3.7.3
Log Directory Writable: ✔
WP Version: 6.7.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.2.27
PHP Post Max Size: 2 GB
PHP Time Limit: 850
PHP Max Input Vars: 6144
cURL Version: 8.11.1
OpenSSL/3.0.15

SUHOSIN Installed: –
MySQL Version: 10.11.10-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 9.6.0
WC Database Prefix: wp_
Total Database Size: 2.91MB
Database Data Size: 1.22MB
Database Index Size: 1.69MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.13MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.03MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2
revision: 1
wp_font_face: 36
wp_font_family: 12

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (6) ###

Query Monitor: by John Blackbourn – 3.17.0
Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.5
Jetpack Beta Tester: by Automattic – 4.1.1
WooCommerce Beta Tester: by WooCommerce – 2.5.1
WooPayments: by WooCommerce – 8.9.0-dev_PR-10135_commit-12e9aad
WooCommerce: by Automattic – 8.9.0-dev-8482717630-g9a50fca

### Inactive Plugins (1) ###

WooCommerce: by Automattic – 9.6.0-beta.1 (update to version 9.7.0-beta.1 is available)

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
db.php: Query Monitor Database Class (Drop-in)
object-cache.php: Memcached

### Must Use Plugins (1) ###

load.php: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 11 KB

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.3
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /srv/htdocs/wp-content/plugins/woocommerce-dev/templates/block-notices/error.php
/srv/htdocs/wp-content/plugins/woocommerce-dev/templates/block-notices/notice.php
/srv/htdocs/wp-content/plugins/woocommerce-dev/templates/block-notices/success.php


### WooPayments ###

Version: 8.9.0-dev_PR-10135_commit-12e9aad
Connected to WPCOM: ❌ No
Logging: Disabled

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
product-pre-publish-modal
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
product-custom-fields
settings
async-product-editor-category-field
launch-your-store
beta-tester-slotfill-examples

Daily Cron: ✔ Next scheduled: 2025-01-29 07:17:15 +00:00
Options: ✔
Notes: 59
Onboarding: skipped

### Action Scheduler ###

Complete: 5
Oldest: 2025-01-28 07:17:21 +0000
Newest: 2025-01-28 07:26:43 +0000

Failed: 1
Oldest: 2025-01-28 07:21:22 +0000
Newest: 2025-01-28 07:21:22 +0000

Pending: 2
Oldest: 2025-01-28 19:26:43 +0000
Newest: 2025-01-29 07:17:21 +0000


### Status report information ###

Generated at: 2025-01-28 08:09:47 +00:00
`

</details>

**Comment by flakas** [comment]
I'd like to get an engineer's eyes on this before we make a call, so I've flagged it for developer feedback. Someone will take a proper look at the repro steps and how it's behaving now.

The bit that needs a closer look: The claim depends on Jetpack Beta Tester plus WooPayments RC activation behavior, and I could not tie that workflow to a current WooCommerce-core code path in trunk. It might be out of scope for this repo.

**Comment by frosso** [comment]
<img width="225" height="578" alt="Image" src="https://github.com/user-attachments/assets/16e6a133-4741-4982-8611-ea40a5ca6083" />

no longer reproducible, and `reactify-classic-payments-settings` is no longer a feature (it's a default). Closing.
